grafana: add google oauth2 config

jwygoda opened the pull request:
###### Motivation for this change

Grafana supports Google OAuth2.
https://grafana.com/docs/grafana/latest/auth/google/

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

aanderse commented on nixos/modules/services/monitoring/grafana.nix:
```diff
@@ -528,23 +532,57 @@ in {
+          default = "";
+          type = types.str;
+        };
+        clientSecret = mkOption {
```
How about we don't add this option.

If the sysadmin is so inclined to ironically treat secrets for this type of application without care, they are always free to do something like:

```
clientSecretFile = pkgs.writeFile "bad-idea.txt" "secret-that-isnt-so-secret";
```

jwygoda replied:
Done

aanderse replied:
That's great. Thank you!

aanderse reviewed (approved):
I don't use `grafana` so I can't test this or anything... but it's a pretty straight forward and LGTM 👍 

Would be nice if someone who knows something about `grafana` could take a quick look before we merge, but ping me if that doesn't happen in the next week or so.

jwygoda commented:
Will do, thanks for the feedback.

jwygoda committed:
grafana: add google oauth2 config

Grafana supports Google OAuth2.
https://grafana.com/docs/grafana/latest/auth/google/

jwygoda commented:
@aanderse

aanderse commented:
@GrahamcOfBorg test grafana

aanderse commented:
Thanks @jwygoda :tada:
